exercises/practice/collatz-conjecture/.approaches/introduction.md

```diff
@@ -0,0 +1,81 @@
+# Introduction
+
+There are at least a couple of idiomatic ways to solve Collatz Conjecture.
+One approach is to use [`filter()`][filter] and [`map()`][map] with [`Stream`][stream] and [`takeWhile()`][takewhile].
+Another approach is to use [`match`][match] with [recursion][recursion].
+
+## Approach `filter()` and `map()` with `Stream()` and `takeWhile()`
+
+```java
+object CollatzConjecture {
+
+  def steps(start: Int): Option[Int] = {
+    Option(start)
+      .filter(_ > 0)
+      .map(collatzStream(_).takeWhile(_ != 1).length)
+  }
+  
+  private def collatzStream(n: Int): Stream[Int] =
+    Stream.iterate(n)(n => if (n % 2 == 0) n / 2 else (n * 3) + 1)
+}
+```
+
+For more information, check the
+[`filter()` and `map()` with `Stream()` and `takeWhile()` approach][approach-filter-map-stream-takewhile].
+
+## Approach: `match` with recursion
+
+```java
+import scala.annotation.tailrec
+
+object CollatzConjecture {
+
+  private object Status extends Enumeration {
+    type Status = Value
+    val ILLEGAL, EVEN, ODD, ONE = Value
+  }
+  import Status._
+
+  private def getStatus(num: Integer): Status = {
+    if (num <= 0)
+      ILLEGAL
+    else if (num == 1)
+      ONE
+    else if (num % 2 == 0)
+      EVEN
+    else
+      ODD
+  }
+
+  @tailrec
+  private def collatzMeBaby(steps: Int, num: Int): Option[Int] = {
```

Comment:
Maybe it's me, but I don't care much for the "Baby" suffix. Maybe some different name?

Reply:
Changed suffix to "Recur". Should be done churning in another half an hour...